cpp/src/utilities/device_atomics.cuh

```diff
@@ -0,0 +1,528 @@
+    return cudf::detail::typesAtomicOperation64
+        (address, val, [](T* a, T v){return atomicMin(a, v);});
+}
+
+// ------------------------------------------------------------------------
+// Overloads for `atomicMax`.
+// Cuda supports `sint32`, `uint32`, `sint64`, `uint64`
+
+__forceinline__ __device__
+int8_t atomicMax(int8_t* address, int8_t val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceMax{});
+}
+
+__forceinline__ __device__
+int16_t atomicMax(int16_t* address, int16_t val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceMax{});
+}
+
+__forceinline__ __device__
+int64_t atomicMax(int64_t* address, int64_t val)
+{
+    using T = long long int;
+    return cudf::detail::typesAtomicOperation64
+        (address, val, [](T* a, T v){return atomicMax(a, v);});
+}
+
+__forceinline__ __device__
+float atomicMax(float* address, float val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceMax{});
+}
+
+__forceinline__ __device__
+double atomicMax(double* address, double val)
+{
+    return cudf::genericAtomicOperation(address, val, cudf::DeviceMax{});
+}
+
+// for wrappers
+inline  __device__
+cudf::date32 atomicMax(cudf::date32* address, cudf::date32 val)
+{
+    using T = int;
+    return cudf::detail::typesAtomicOperation32
+        (address, val, [](T* a, T v){return atomicMax(a, v);});
+}
+
+__forceinline__ __device__
+cudf::category atomicMax(cudf::category* address, cudf::category val)
+{
+    using T = int;
+    return cudf::detail::typesAtomicOperation32
+        (address, val, [](T* a, T v){return atomicMax(a, v);});
+}
+
+__forceinline__ __device__
+cudf::date64 atomicMax(cudf::date64* address, cudf::date64 val)
+{
+    using T = long long int;
+    return cudf::detail::typesAtomicOperation64
+        (address, val, [](T* a, T v){return atomicMax(a, v);});
+}
+
+__forceinline__ __device__
+cudf::timestamp atomicMax(cudf::timestamp* address, cudf::timestamp val)
+{
+    using T = long long int;
+    return cudf::detail::typesAtomicOperation64
+        (address, val, [](T* a, T v){return atomicMax(a, v);});
+}
+
+// ------------------------------------------------------------------------
+// Overloads for `atomicCAS`.
+// Cuda supports `sint32`, `uint32`, `uint64`
+// int8_t, int16_t are never supported as overloads
+
+__forceinline__ __device__
+int64_t atomicCAS(int64_t* address, int64_t compare, int64_t val)
+{
+    using T = int64_t;
+    return cudf::detail::typesAtomicCASImpl<T, sizeof(T)>()(address, compare, val);
+}
+
+__forceinline__ __device__
+float atomicCAS(float* address, float compare, float val)
+{
+    using T = float;
+    return cudf::detail::typesAtomicCASImpl<T, sizeof(T)>()(address, compare, val);
+}
+
+__forceinline__ __device__
+double atomicCAS(double* address, double compare, double val)
+{
+    using T = double;
+    return cudf::detail::typesAtomicCASImpl<T, sizeof(T)>()(address, compare, val);
+}
+
+// for wrappers
```

Comment:
Can you clarify this comment?

Reply:
It means below functions are for cudf `wrapper` types.
I will remove this comment because each overload has been doxymentation